libhb/audio_resample.c

```diff
@@ -85,6 +105,11 @@ void hb_audio_resample_set_channel_layout(hb_audio_resample_t *resample,
             // Dolby Surround is Stereo when it comes to remixing
             channel_layout = AV_CH_LAYOUT_STEREO;
         }
+        if (resample->out.channel_layout == AV_CH_LAYOUT_MONO &&
+            is_mono(channel_layout))
+        {
+            resample->out.channel_layout = channel_layout;
+        }
         resample->in.channel_layout = channel_layout;
     }
```

Comment:
Not a big deal, but technically, out.channel_layout is set once during init ands fixed for the whole encode, whereas in.channel_layout may vary from frame to frame. So it would be more correct to set in.channel_layout = MONO.

Also a one-line comment would be useful, like:

// avresample can't remap a single-channel layouts to another single-channel layout

Otherwise, LGTM